<!--
  Standard launch configuration for an app manager used as a concert client.
  This provides meaningful arg names and fixes some args such as the firewall which
  shouldn't be modified in a concert setting.
 -->
<launch>
  <!-- ********************************* Robot ********************************* -->
  <arg name="robot_description"      default="To err is human, to 'arr is pirate."/>
  <arg name="robot_name"             default="Cybernetic Pirate"/>
  <arg name="robot_icon"             default="rocon_icons/cybernetic_pirate.png"/> 
  <arg name="robot_type"             default="turtlebot"/>
  <arg name="robot_unique_name"      default="true"/> <!-- postfix a uuid to the robot name for uniqueness -->

  <!-- ********************************* Rapps ********************************* -->
  <arg name="rapp_auto_installation" default="false"/>        <!-- http://wiki.ros.org/rocon_app_manager/Tutorials/indigo/Automatic Rapp Installation -->
  <arg name="rapp_package_whitelist" default="[rocon_apps]"/> <!-- comma separated list of package names -->
  <arg name="rapp_package_blacklist" default="[]"/>

  <!-- ******************************* Connectivity ***************************** -->
  <arg name="concert_whitelist"     default="[]"/>    <!-- list of concert names this robot will work with -->
  <arg name="local_machine_only"    default="false"/> <!-- only work with local concerts (testing, simulations) -->
  <arg name="concert_watch_period"  default="10"/>    <!-- the period used by gateways for watching concert connections -->

  <!-- ******************************* Debugging ******************************** -->
  <arg name="screen"                 default="false"/> <!-- verbose output from running apps -->

  <!-- **************************** Optional Modules **************************** -->
  <arg name="capabilities"                   default="false"/> <!-- enables/disables a default capability server in this concert client -->
  <arg name="capabilities_blacklist"         default="[]" />   <!-- blacklist specific capabilities -->
  <arg name="capabilities_package_whitelist" default="[]" />   <!-- get capabilities from these packages only (e.g. std_capabilities) -->
  <arg name="capabilities_parameters"        default="$(find rocon_app_manager)/param/capabilities.yaml" /> <!-- detailed parameter configuration for the providers -->

  <arg name="interactions"          default="false"/>
  <arg name="interactions_list"     default="[]"/>

  <!-- ************************* Rarely Used Variables ************************** -->
  <arg name="firewall" default="false"/> <!-- typically false (don't let anything in), only for simulation clients -->
  
  <!-- ******************************* Launcher ********************************* -->

  <include file="$(find rocon_app_manager)/launch/multimaster.launch">
    <arg name="auto_rapp_installation"         value="$(arg rapp_auto_installation)" />
    <arg name="auto_start_rapp"                value="false" />
    <arg name="capabilities"                   value="$(arg capabilities)" />
    <arg name="capabilities_blacklist"         value="$(arg capabilities_blacklist)" />
    <arg name="capabilities_package_whitelist" value="$(arg capabilities_package_whitelist)" />
    <arg name="capabilities_parameters"        value="$(arg capabilities_parameters)" />
    <arg name="interactions"                   value="$(arg interactions)"/>
    <arg name="interactions_list"              value="$(arg interactions_list)"/>
    <arg name="gateway_firewall"               value="$(arg firewall)"/>
    <arg name="gateway_uuids"                  value="$(arg robot_unique_name)" />
    <arg name="gateway_watch_loop_period"      value="$(arg concert_watch_period)" />
    <arg name="hub_whitelist"                  value="$(arg concert_whitelist)"/>
    <arg name="local_remote_controllers_only"  value="$(arg local_machine_only)" />
    <arg name="rapp_package_whitelist"         value="$(arg rapp_package_whitelist)" />
    <arg name="rapp_package_blacklist"         value="$(arg rapp_package_blacklist)" />
    <arg name="robot_description"              value="$(arg robot_description)" />
    <arg name="robot_name"                     value="$(arg robot_name)" />
    <arg name="robot_icon"                     value="$(arg robot_icon)" />
    <arg name="robot_type"                     value="$(arg robot_type)" />
    <arg name="screen"                         value="$(arg screen)" />
  </include>

</launch>
